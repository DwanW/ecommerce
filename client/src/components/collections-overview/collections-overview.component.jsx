import React from 'react';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { selectColletionsForPreview } from '../../redux/shop/shop.selector';

import {CollectionsOverviewContainer} from './collections-overview.styles';

import CollectionPreview from '../collection-preview/collection-preview.component';


export const CollectionsOverview = ({ collections }) => (
    <CollectionsOverviewContainer>
        {
            collections.map(({ id, ...otherCollectionProps }) => (
                <CollectionPreview key={id} {...otherCollectionProps} />
            ))
        }
    </CollectionsOverviewContainer>
)

const mapStateToProps = createStructuredSelector({
    collections: selectColletionsForPreview
})

export default connect(mapStateToProps)(CollectionsOverview);
