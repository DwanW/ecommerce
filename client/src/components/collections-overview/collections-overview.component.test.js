import React from 'react';
import { shallow } from 'enzyme';
import {CollectionsOverview} from './collections-overview.component';

describe('CollectionsOverview test', () => {
    const wrapper = shallow(<CollectionsOverview collections={[]}/>);
    it('should render CollectionsOverview component', ()=> {
        expect(wrapper).toMatchSnapshot();
    });

    it('should render no child with [] collections', ()=> {
        expect(wrapper.find('CollectionsOverviewContainer').children().length).toBe(0)
    });
});