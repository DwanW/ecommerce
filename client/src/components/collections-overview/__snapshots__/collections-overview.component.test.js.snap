// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollectionsOverview test should render CollectionsOverview component 1`] = `ShallowWrapper {}`;
